aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/FluentClientGenerator.kt

```diff
@@ -114,8 +114,11 @@ class FluentClientGenerator(protocolConfig: ProtocolConfig) {
             rustTemplate(
                 """
                 ##[cfg(any(feature = "rustls", feature = "native-tls"))]
-                pub fn from_env() -> Self {
-                    Self::from_conf(crate::Config::builder().build())
+                pub async fn from_env() -> Self {
+                    // backwards compatibility shim
```

Comment:
Is this accomplishing backwards compatibility if we're going from `fn` to `async fn`?

Reply:
backwards compatible in behavior but not literally